### PR TITLE
update github actions to use latest module template versions

### PR DESCRIPTION
**Pull request by adonesky1**
The GitHub actions have been updated to match the module template.

There are still differences because this library is not yet using TypeScript, so it doesn't have any build step.

It also doesn't yet have API documentation, so steps relating to that have been removed.

### CODE REVIEW

File: .github/workflows/publish-release.yml [hunk 1]
```diff
@@ -1,29 +1,54 @@
...
+        with:
+          ref: ${{ github.sha }}
+      - name: Publish
+        uses: MetaMask/action-npm-publish@v2
+        with:
+          # This `NPM_TOKEN` needs to be manually set per-repository.
+          # Look in the repository settings under "Environments", and set this token in the `npm-publish` environment.
+          npm-token: ${{ secrets.NPM_TOKEN }}
+        env:
+          SKIP_PREPACK: true
```
[the remark this answers is not in the document]
Comment (Gudahtt) [reply .github/workflows/publish-release.yml:54]: I have forgotten this step _at least_ twice now. We should at the very least add a comment on this line of the module template to remind ourselves of this in the future.

[the remark this answers is not in the document]
Comment (adonesky1) [reply .github/workflows/publish-release.yml:54]: done [here](https://github.com/MetaMask/eth-hd-keyring/pull/73/commits/c4aeba4bb7f1c88e5d9e9970ce222c0dd5422e94)

File: package.json [hunk 2]
```diff
@@ -22,6 +22,7 @@
     "lint:eslint": "eslint . --cache --ext js,ts",
     "lint:fix": "yarn lint:eslint --fix && yarn lint:misc --write",
     "lint:misc": "prettier '**/*.json' '**/*.md' '!CHANGELOG.md' '**/*.yml' '!.yarnrc.yml' --ignore-path .gitignore --no-error-on-unmatched-pattern",
+    "prepack": "chmod +x ./scripts/prepack.sh",
```
Comment (Gudahtt) [inline package.json:25]: chmod only needs to be run once locally to set the file permissions correctly. The file permissions are stored in git.

This script should just invoke the prepack script.
```suggestion
    "prepack": "./scripts/prepack.sh",
```

File: scripts/prepack.sh [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+set -x
+set -e
+set -o pipefail
+
+if [[ -n $SKIP_PREPACK ]]; then
+  echo "Notice: skipping prepack."
+  exit 0
+fi
+
+yarn build
```
Comment (Gudahtt) [inline scripts/prepack.sh:12]: Wait a minute... there is no build step here! So maybe this wasn't needed. Huh.

Sorry about that, I hadn't considered that there was no prepack npm script to migrate here. Feel free to revert this addition, or remove this build command leaving this around as a placeholder for when we do finally migrate to TypeScript

Comment (adonesky1) [reply scripts/prepack.sh:12]: haha yep whoops I clearly wasn't thinking about it either!

Comment (adonesky1) [reply scripts/prepack.sh]: left it as a placeholder but removed build step [here](https://github.com/MetaMask/eth-hd-keyring/pull/73/commits/84cc6cee6a07e2661bc997d4390ae38837d79acd)

File: package.json (hunk 2 above)
Comment (adonesky1) [reply package.json]: done [here](https://github.com/MetaMask/eth-hd-keyring/pull/73/commits/84cc6cee6a07e2661bc997d4390ae38837d79acd)

**Review by Gudahtt — APPROVED** [review]
LGTM!

**Pull request merged**